File: SegundoSemestre/Java/Clase_11/Persona.java
//  Clase 11 Parte 1
//  Video 4

public class Persona {
    private String nombre;

    public Persona(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    //  Video 5

    @Override
    public String toString() {
        return "Persona{" + "nombre=" + nombre + '}'+", "+super.toString();//este toString es de la clase padre Object
    }
    
}

File: SegundoSemestre/Python/Leccion2/Clase_11/Persona.py
#   CLASE 11 Parte 1
#   VIDEO 8
class Persona:
    def __init__(self,nombre,edad):
        self.nombre = nombre
        self.edad = edad
        
    def __add__(self, other): # other = otro
        return f'{self.nombre} {other.nombre}'
    
    #   VIDEO 9
    def __sub__(self, otro):
        return self.edad - otro.edad
    
persona1 = Persona('Ariel', 40)
persona2 = Persona('Betancud', 5)

#persona1.__add__(persona2) # Sintaxis interna y automatica
print(persona1 + persona2) # Sintaxis normal

#   VIDEO 9
print(persona1 - persona2)